gcloud-java-storage/src/main/java/com/google/gcloud/storage/ObjectId.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.google.gcloud.storage;
+
+import java.net.URI;
+import java.util.Objects;
+
+/**
+ * Identifier for a stored object.
+ */
+public class ObjectId {
```

Comment:
I think we should agree how we plan to use it (issue #62) before adding it should probably effect its implementation and signature.
e.g.
As discussed in issue #62 my view on it is, I think, closer to Amazon implementation where it is used more as a tool to extract bucket and blob names from a URI. The rest of the API is using bucket & name/key for object reference.
In that case I would not call it ObjectId but rather something like that better describe its intent (also, in that case it could also be a static method to construct a BlobInfo.Builder given a URI).